tests/test_result_conversion.rs

```diff
@@ -0,0 +1,45 @@
+//! Testing https://github.com/PyO3/pyo3/issues/1106. A result type that
+//! implements `From<MyError> for PyErr` should be automatically converted
+//! when using `#[pyfunction]`.
+//!
+//! This only makes sure that valid `Result` types do work. For an invalid
+//! enum type, see `ui/invalid_result_conversion.py`.
+use pyo3::prelude::*;
+use pyo3::wrap_pyfunction;
+
+use std::fmt;
+
+mod common;
+
+/// A basic error type for the tests.
+#[derive(Debug)]
+struct MyError {
+    pub descr: String,
```

Comment:
Just a nit, but this should be `&'static str`

Reply:
This is something I never know how to deal with. Why exactly `&'static str`? Isn't it better to use `String` in structs and similars because they're owned types and that way you avoid lifetimes issues, like [clap](https://github.com/clap-rs/clap/issues/1041) (although they prefer to use `Cow` for that).

Reply:
In this case we're just hard-coding an example so there's no need to take ownership or take a lifetime more flexible than `'static`. (You're right in most cases you'd want `&'a str` or an owned `String`.)

Reply:
Since this test don't have to be too general.